packages/network-controller/src/create-network-client.ts

```diff
@@ -136,17 +142,21 @@ export function createNetworkClient({
     });
   });
 
-  const rpcApiMiddleware =
-    configuration.type === NetworkClientType.Infura
-      ? createInfuraMiddleware({
-          rpcService: rpcServiceChain,
-          options: {
-            source: 'metamask',
-          },
-        })
-      : createFetchMiddleware({ rpcService: rpcServiceChain });
+  let rpcApiMiddleware: RpcApiMiddleware;
+  if (configuration.type === NetworkClientType.Infura) {
+    rpcApiMiddleware = asV2Middleware(
+      createInfuraMiddleware({
+        rpcService: rpcServiceChain,
+        options: {
+          source: 'metamask',
+        },
+      }),
+    ) as unknown as RpcApiMiddleware;
```

Comment:
I will try to improve the type of `asV2Middleware` to get rid of this and other casts we perform on its return values.

Reply:
Got rid of this one, at least.